### PR TITLE
bpo-39495: Remove default value from C impl of TreeBuilder.start

### PR DESCRIPTION
**Pull request by hauntsaninja**
https://bugs.python.org/issue39495

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2020-01-30-07-02-02.bpo-39495.8LsIRN.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove default value from ``attrs`` parameter of ``xml.etree.ElementTree.TreeBuilder.start`` for consistency between Python and C implementations.
```
Comment (corona10) [inline Misc/NEWS.d/next/Library/2020-01-30-07-02-02.bpo-39495.8LsIRN.rst]:
```suggestion
Remove default value from *attrs* parameter of :meth:`xml.etree.ElementTree.TreeBuilder.start` for consistency between Python and C implementations.
```

**Review by corona10 — CHANGES_REQUESTED** [review]

**Comment by serhiy-storchaka** [comment]
Thank you for your PR @hauntsaninja.

I prefer to remove also the support of None as a value of `attrib` in `treebuilder_handle_start`. According to the documentation `attrib` is a dict, and a dict is always passed to custom `TreeBuilder.start()`.

`treebuilder_handle_start` is called from two places: the implementation of `TreeBuilder.start()` and `expat_start_handler`. In the latter case None is passed for optimization, to avoid creation of an empty dictionary. We can use NULL instead of None for this optimization.

**Comment by serhiy-storchaka** [comment]
Or you can just use `object(subclass_of='&PyDict_Type')` for `attrs`.

**Comment by hauntsaninja** [comment]
Thanks Dong-hee and Serhiy! Don't know much about argument clinic converters, that's a neat fix.

**Review by corona10 — CHANGES_REQUESTED** [review]
Looks good.
I'd like to add unit tests for this change.
You can add in these files.
- https://github.com/python/cpython/blob/master/Lib/test/test_xml_etree.py
- https://github.com/python/cpython/blob/master/Lib/test/test_xml_etree_c.py

**Comment by hauntsaninja** [comment]
Thanks, sorry for the delay, done. It seems test_xml_etree_c.py automatically runs all tests in test_xml_etree.py, so I only added a test for the latter.

**Review by corona10 — APPROVED** [review]

**Comment by corona10** [comment]
@serhiy-storchaka Can you please take a look?

File: Lib/test/test_xml_etree.py [hunk 2]
```diff
@@ -2259,6 +2259,9 @@ def test_expat224_utf8_bug_file(self):
         text = text[6:-4]
         self.assertEqual(root.get('b'), text)
 
+    def test_39495_treebuilder_start(self):
+        self.assertRaises(TypeError, ET.TreeBuilder().start, "tag", None)
```
Comment (serhiy-storchaka) [inline Lib/test/test_xml_etree.py:2264]: Please add also a test for `start()` with a single argument.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**